spec/features/global_announcements_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe 'Global announcements', type: :feature do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    describe 'no global announcements' do
+      before do
+        instance.announcements.clear
```

Comment:
Yes. All these are best practices. Best practices that, IMO, are being applied without considering the circumstances. Engineering is not only about following rules. It's knowing when exceptions need to be made when other considerations are in play.